File: problems/0100/0001/0001.py


from ast import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        sub_dict = {}
        for i in range(len(nums)):
            sub = target - nums[i]
            if sub in sub_dict:
                return [sub_dict[sub], i]
            else:
                sub_dict[nums[i]] = i

        return []

File: problems/0100/0001/_0001.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] result = new int[2];

        Map<Integer, Integer> subMap = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int sub = target - nums[i];
            if (subMap.containsKey(sub)) {
                result[0] = subMap.get(sub);
                result[1] = i;
                break;
            } else {
                subMap.put(nums[i], i);
            }
        }

        return result;
    }
}
